cache.go

```diff
@@ -190,32 +190,51 @@ func newCache(
 
 // NewCacheByRequestURI a shortcut function for caching response by uri
 func NewCacheByRequestURI(defaultCacheStore persist.CacheStore, defaultExpire time.Duration, opts ...Option) app.HandlerFunc {
-	options := newOptions(opts...)
+	cacheStrategy := func(ctx context.Context, c *app.RequestContext) (bool, Strategy) {
+		return true, Strategy{
+			CacheKey: string(c.Request.RequestURI()),
+		}
+	}
+	var o []Option
+	o = append(o, WithCacheStrategyByRequest(cacheStrategy))
+	o = append(o, opts...)
+	return NewCache(defaultCacheStore, defaultExpire, o...)
+}
 
-	var cacheStrategy GetCacheStrategyByRequest
-	if options.ignoreQueryOrder {
-		cacheStrategy = func(ctx context.Context, c *app.RequestContext) (bool, Strategy) {
-			newUri, err := getRequestUriIgnoreQueryOrder(c.Request.URI().String())
-			if err != nil {
-				hlog.CtxErrorf(ctx, getRequestUriIgnoreQueryOrderErrorFormat, err)
-				newUri = c.Request.URI().String()
-			}
+// NewCacheByRequestURIWithIgnoreQueryOrder a shortcut function for caching response by uri and ignore query param order.
+func NewCacheByRequestURIWithIgnoreQueryOrder(defaultCacheStore persist.CacheStore, defaultExpire time.Duration, opts ...Option) app.HandlerFunc {
+	cacheStrategy := func(ctx context.Context, c *app.RequestContext) (bool, Strategy) {
+		newUri, err := getRequestUriIgnoreQueryOrder(string(c.Request.RequestURI()))
```

Comment:
Why use `.RequestURI()` instead of the `.URI()`?

Reply:
Do not need Scheme and Host anymore?

Reply:
Yes, aligned with Gin, Gin's URI is `/a/b?a=1`, but hertz is not. Many migration users assume that `c.Request.URI().String()` and gin behave the same way, but they don't.